geode-core/src/main/java/org/apache/geode/distributed/internal/ClusterDistributionManager.java

```diff
@@ -2821,4 +2827,24 @@ public CancelCriterion getCancelCriterion() {
     return stopper;
   }
 
+  static class ClusterDistributionManagerIDFactory implements MemberIdentifierFactory {
+    @Immutable
+    private static final Comparator<MemberIdentifier> idComparator = new Comparator() {
+      @Override
+      public int compare(Object o1, Object o2) {
+        return ((DistributedMember) o1).compareTo((DistributedMember) o2);
+      }
+    };
+
+    @Override
+    public MemberIdentifier create(MemberData memberInfo) {
+      return new InternalDistributedMember(memberInfo);
+    }
+
+    @Override
+    public Comparator<MemberIdentifier> getComparator() {
```

Comment:
Could we just make MemberIdentifier extend comparable?

Reply:
what's wrong the the compositional approach here @upthewaterspout ?

Reply:
Ok, makes sense.